server/src/main/java/org/elasticsearch/cluster/routing/allocation/AllocationService.java

```diff
@@ -644,6 +645,11 @@ private boolean assertInitialized() {
         return true;
     }
 
+    // exposed for tests whose behaviour depends on this
+    boolean isBalancedShardsAllocator() {
```

Comment:
We could reasonably port this (and the places that depend on it) to `main` too, reducing the size of the branch merge.